sw/device/lib/dif/dif_alert_handler_unittest.h

```diff
@@ -0,0 +1,40 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_ALERT_HANDLER_UNITTEST_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_ALERT_HANDLER_UNITTEST_H_
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/base/testing/mock_mmio.h"
+#include "sw/device/lib/dif/dif_alert_handler.h"
+
+#include "alert_handler_regs.h"  // Generated.
+
+namespace dif_alert_handler_unittest {
+namespace {
+
+using ::mock_mmio::LeInt;
+using ::mock_mmio::MmioTest;
+using ::mock_mmio::MockDevice;
+using ::testing::_;
+using ::testing::Return;
+
+constexpr uint32_t kAlerts = ALERT_HANDLER_PARAM_N_ALERTS;
+constexpr uint32_t kAllZeros = 0;
+
+class AlertHandlerTest : public testing::Test, public MmioTest {
```

Comment:
This seems too much of a convention, but I am willing to concede. Why not just have the autogen code rely on its own base test class?

Reply:
oooo, yes silly me, I should've done that from the start then we don't need a *_unittest.h as depicted in #8142. This is much cleaner (and easier on me in the long run!). I will address this fix in #8304 and rebase on top, since its similar (just integrating autogen IRQ DIFs into HMAC instead of Alert Handler) and a requested fix in #8304 also effects **all** autogen'd IRQ DIF--IP integrations.

Reply:
fixed.